### ISSUE TITLE
False negative from JavadocMethod when a record canonical constructor is documented inside an interface

### ISSUE THREAD

**Issue by jg-partisia** [checkstyle/checkstyle#12200]
I have read check documentation: https://checkstyle.org/config_javadoc.html#JavadocMethod
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
$ javac Interface.java

$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocMethod"/>
  </module>
</module>

$ cat Interface.java
package my.pack;

import static java.util.Objects.requireNonNull;

/** My outer interface. */
public interface Interface {

  /** My inner record. */
  public record Record(int field) {
    /**
     * Constructor.
     *
     * @param field Some field.
     */
    public Record {
      requireNonNull(field);
    }
  }
}

$ RUN_LOCALE="-Duser.language=en -Duser.country=US"
$ java $RUN_LOCALE -jar checkstyle-10.3.3-all.jar -c config.xml Interface.java
Starting audit...
[ERROR] C:\Users\jesper\tmp\checkstyle_bug\Interface.java:13: Unused Javadoc tag. [JavadocMethod]
Audit done.
Checkstyle ends with 1 errors.

$ cat OtherRecord.java
package my.pack;

import static java.util.Objects.requireNonNull;

/** My other record. */
public record OtherRecord(int field) {
  /**
   * Constructor.
   *
   * @param field Some field.
   */
  public OtherRecord {
    requireNonNull(field);
  }
}

$ java $RUN_LOCALE -jar checkstyle-10.3.3-all.jar -c config.xml OtherRecord.java
Starting audit...
Audit done.
```

---

I expect checkstyle to accept the javadoc for the record constructor - just it does when the record is not placed inside an interface.

**Comment by tiller** [comment]
I believe this has been fixed somewhere between 10.17.0 and 10.23.0 and can be closed

**Comment by romani** [comment]
Fixed by https://github.com/checkstyle/checkstyle/issues/15683

https://checkstyle.org/releasenotes.html#Release_10.21.3
